### DISCUSSION TITLE
Performances et chargement des éléments

### DISCUSSION THREAD

**Discussion by lucmuller** [AmauriC/tarteaucitron.js#1053] [Ideas]
J'ouvre cette discussion, non pas pour demander un changement rapide, mais pour ouvrir une réflexion relative à l'usage que j'ai de Tarte Au Citron.

1) Chargement des services 
A l'heure actuelle, TAC possède une configuration pour activer/désactiver des services via des configurations.

`(tarteaucitron.job = tarteaucitron.job || []).push('service');`

Tous les services sont compilés dans un seul fichier Javascript qui est servit par le site.
*Problème :* Les sites, en général, n'utilisent qu'une fraction réduite des services ce qui entraine le suivant.
Dès qu'une équipe analysant les performances, elle recommande de supprimer les services inutiles. Ce qui implique de ne plus utiliser le chargement automatique de TAC.

=> Est-ce qu'il ne serait pas plus pertinent de ne pas avoir de fichier tarteaucitron.services.js mais plutôt un fichier "text" d'exemple, charge à l'intégrateur de choisir les services qu'il souhaites intégrer.
De fait, un service étant intégré/configuré impliquerai qu'il soit actif.

Le fait d'avoir "plein" de services donne aussi l'impression que TAC est une solution _exhaustive_ qui si un service n'est pas disponible n'apporte pas de solution pour le service manquant. Peut-être que cela pourrait donner envie à des gens de proposer leurs propres services qui pourraient alors être combinés manuellement par les intégrateurs.

2) Minification des ressources.
Lorsque l'on utilise un système de minification des ressources, les fichiers sont souvent généré à un endroit spécifique qui n'est pas celui où se trouve la ressource originelle
Aussi, si le fichier de TAC est minifié et que le fichier de service est chargé dynamiquement, sont chargement est fait relativement à l'emplacement du fichier js minifié et souvent cela résulte en une 404 sur le fichier en question.
La conséquence est la même qu'au point 1 pour minifier correctement les fichiers, on ne peut pas utiliser le chargement automatique.

Il en va de même de la feuille de style. En travaillant avec scss, j'importe la css de TAC dans mes fichiers scss que je surcharge. Le tout est minifié concaténé etc.

Au final, dans les derniers projets que j'ai monté, je charge systématiquement toutes les ressources de TAC manuellement.

De mon point de vue des choses pourraient être améliorer à ce niveau et les chargements "automatiques" mis de côté.

J'ai fini, merci de m'avoir écouté ^^

**Comment by initmaster** [comment]
Je plussoie. Proposer les différents services dans des fichiers distincts en plus du fichier tarteaucitron.services.js serait un grand plus pour que les intégrateurs puissent selectionner manuellement les services dont ils ont besoin (La plupart du temps je n'en ai que 2 voir 5 grand max).

Pour ma part j'ai dans le cas de ressources minifiés a un autre endroit j'ai utilisé l'option tarteaucitron.cdn et y ai inclu le path vers le bon repertoire. Un plus serait d'avoir une option pour indiquer au script de charger des versions minifiés du fichier de langue et des services.
